modules/phase_field/src/materials/OPInterfaceBarrierMaterial.C

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "OPInterfaceBarrierMaterial.h"
+
+template<>
+InputParameters validParams<OPInterfaceBarrierMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addParam<std::string>("function_name", "g", "actual name for g(eta_i)");
+  MooseEnum h_order("SIMPLE=0", "SIMPLE");
+  params.addParam<MooseEnum>("g_order", h_order, "Polynomial order of the switching function h(eta)");
+  params.addRequiredCoupledVar("etas", "eta_i order parameters, one for each h");
+
+  return params;
+}
+
+OPInterfaceBarrierMaterial::OPInterfaceBarrierMaterial(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+    _function_name(getParam<std::string>("function_name")),
+    _g_order(getParam<MooseEnum>("g_order")),
+    _num_eta(coupledComponents("etas")),
+    _eta(_num_eta),
+    _prop_g(declareProperty<Real>(_function_name)),
+    _prop_dg(_num_eta),
+    _prop_d2g(_num_eta)
+
+{
+
+  for (unsigned int i = 0; i < _num_eta; ++i)
+    _prop_d2g[i].resize(_num_eta);
+
+  // declare derivative properties, fetch eta values
+  std::vector<std::string> eta_name(_num_eta);
+  for (unsigned int i = 0; i < _num_eta; ++i)
+    eta_name[i] = getVar("etas", i)->name();
+
+  for (unsigned int i = 0; i < _num_eta; ++i)
+  {
+    _prop_dg[i]  = &declarePropertyDerivative<Real>(_function_name, eta_name[i]);
+    _eta[i] = &coupledValue("etas", i);
+    for (unsigned int j = i; j < _num_eta; ++j)
+    {
+      _prop_d2g[i][j] =
```

Comment:
Actually I don't know your physics well enough to say for certain and there isn't a comment here... Ignoring whitespace this is what you've written:

```
_prop_d2g[i][j] = _prop_d2g[j][i] = &declarePropertyDerivative<Real>(_function_name, eta_name[i], eta_name[j]);
```

So if you are intending to build a symmetric matrix this is correct, still probably worth a comment just to be sure, also the parenthesis aren't necessary. This is actually a single line of code.